### DISCUSSION TITLE
Any affordable DDOS / Bot Protection services to use with Vercel?

### DISCUSSION THREAD

**Discussion by jinsley8** [vercel/vercel#9418] [Help]
Is Cloudflare proxy still not recommended to use with Vercel?

I'm looking for a DDOS/Bot Protection service that will integrate but the two recommended (DataDome and Botd) by Vercel are extremely expensive.

Any other alternatives?

**Comment by leerob** [comment]
Hey, wanted to follow up here.

We just launched **[Attack Challenge Mode](https://vercel.com/changelog/prevent-malicious-traffic-with-attack-challenge-mode-for-vercel-firewall)** –an extra layer of defense for DDoS and malicious traffic, free for all plans. When enabled, the Vercel Firewall will serve a challenge page to help mitigate attacks.

Hope this helps!

**Reply by jinsley8** [reply]
@leerob so this is like DDOS mode on other services where you only enable it when there is an attack to mitigate?

Is there any other way to enable notifications for when traffic or serverless function usage increases unexpectedly or past a threshold?
